tests/PHPStan/Analyser/data/bug-2471.php

```diff
@@ -0,0 +1,19 @@
+<?php
+
+namespace Bug2471;
+
+use function PHPStan\Testing\assertType;
+
+class A {
+	public function __toString()
+	{
+		return 'foo';
+	}
+}
+
+/** @var A[] */
+$y = doFoo();
+
+$x = array_fill_keys($y, null);
+
+assertType('array<string, null>', $x);
```

Comment:
there should not be any code at the top level of a test. please move it into a method (same in other test files)

Reply:
Sure, but can you explain me why ?